File: SSServer/src/main/java/com/prose/crhen/SSServer/business/api/ServerService.java
package com.prose.crhen.SSServer.business.api;

import com.prose.crhen.SSServer.dto.*;
import com.prose.crhen.SSServer.model.Server;

import java.util.List;

public interface ServerService {
    //ONLY FOR DEMONSTRATION
    void deleteAll();

    ServerOverviewDTO getServerOverviewDTO();

    List<ServerQueryDTO> getServers();

    void updateVolume(VolumeQueryDTO volumeQueryDTO);

    void updateServer(ServerQueryDTO serverQueryDTO);

    List<VolumeQueryDTO> getVolumesFromServer(String serverName);

    void saveServerDTO(ServerUpdateDTO server);

    void saveVolumeDTO(VolumesUpdateDTO newServer);

    ServerQueryDTO.ServerQueryDTOBuilder calcServerDetails(Server server);

    List<VolumeQueryDTO> getVolumes();
}

File: SSKotlinServer/src/main/kotlin/com/prose/crhen/SSKotlinServer/SSKotlinServer/business/api/ServerService.kt
package com.prose.crhen.SSKotlinServer.SSKotlinServer.business.api

import com.prose.crhen.SSKotlinServer.SSKotlinServer.dto.*
import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.Server

interface ServerService {
    fun deleteAll();

    fun getServerOverviewDTO(): ServerOverviewDTO

    fun getServers(): List<ServerQueryDTO>

    fun updateVolume(volumeQueryDTO: VolumeQueryDTO)

    fun updateServer(serverQueryDTO: ServerQueryDTO)

    fun getVolumesFromServer(serverName: String): List<VolumeQueryDTO>

    fun saveServerDTO(server: ServerUpdateDTO)

    fun saveVolumeDTO(newServer: VolumesUpdateDTO)

    fun calcServerDetails(server: Server): ServerQueryDTO

    fun getVolumes(): List<VolumeQueryDTO>


}
